tools/wptrunner/wptrunner/wptrunner.py

```diff
@@ -47,7 +49,9 @@ def setup_logging(*args, **kwargs):
     return logger
 
 
-def get_loader(test_paths, product, **kwargs):
+def get_loader(test_paths: Mapping[str, List[str]],
```

Comment:
So, it turns out that Python 3.7 doesn't have `TypedDict`, so I've gone down the rabbit hole of trying to make this into an actual class instead. That's all in a new commit, so @jonathan-j-lee if you don't mind reviewing this again, it would be appreciated (hopefully I found all the corner cases now :) ).

Reply:
> So, it turns out that Python 3.7 doesn't have TypedDict

Ah, that's too bad; I would have been fine with an `Any` placeholder, had I known.

`TestRoot` class LGTM, thanks!